app/src/main/java/com/yalta/repositories/RepoResponse.kt

```diff
@@ -0,0 +1,20 @@
+package com.yalta.repositories
+
+sealed class RepoResponse<T>
+open class SuccessfulResponse<T> : RepoResponse<T>()
+class FailedResponse<T>: RepoResponse<T>()
+
+inline fun <reified S : SuccessfulResponse<S>, R> process(
```

Comment:
Maybe rework this for a better _API_...
Something like 
```
object Request<S,R>{
  object RequestWithAction<S, R>(val action:...){
     object RequestWithSuccess<S,R>(val action, onSuccess){
        fun or(onFail){
           //logic goes here
        }
     }
     fun onSuccess(onSuccess:...){
         return RequestWithSuccess(action, onSuccess)
     }
  }
  fun apply(action:()->RR<S>){
     return RequestWithAction(action)
  }
}
```
and instead of 
```
process(
            { repo.sendCurrentLocation(location) },
            { it.location },
            { null }
        )
```
use this as 
```
Request{ repo.sendCurrentLocation(location) }
 .onSuccess {it.location}
 .or {null}
```

Reply:
Maybe later
Sil'no later...